View timeline item: Contact

murilovmachado opened the pull request:
Jira: https://bugs.benetech.org/browse/CHI-150

This PR implements clicking `View` on a Connected Contact from the timeline.

WIP:
- ~Add more tests~ (done)

<img width="681" alt="Screen Shot 2020-08-18 at 14 08 42" src="https://user-images.githubusercontent.com/1504544/90543657-679ffb00-e15c-11ea-9479-f9c3177864b6.png">
<img width="688" alt="Screen Shot 2020-08-18 at 14 08 29" src="https://user-images.githubusercontent.com/1504544/90543674-6d95dc00-e15c-11ea-8db4-357edaddc844.png">

GPaoloni commented on plugin-hrm-form/src/components/case/AddHousehold.tsx:
```diff
@@ -81,7 +81,10 @@ const AddHousehold: React.FC<Props> = ({
         <ActionHeader titleTemplate="Case-AddHousehold" onClickClose={onClickClose} counselor={counselor} />
-        <CallerForm callerInformation={temporaryCaseInfo} defaultEventHandlers={defaultEventHandlers} />
+        <CallerForm
+          callerInformation={temporaryCaseInfo as CallerFormInformation}
```
Above suggestion also removes the need for this cast.

GPaoloni commented on plugin-hrm-form/src/components/case/AddPerpetrator.tsx:
```diff
@@ -59,7 +59,7 @@ const AddPerpetrator: React.FC<Props> = ({
 
     const { info } = connectedCaseState.connectedCase;
-    const perpetrator = getFormValues(temporaryCaseInfo);
+    const perpetrator = getFormValues(temporaryCaseInfo) as FormValues<CallerFormInformation>;
```
The two comments in the previous file apply for this and below casting too.

GPaoloni commented on plugin-hrm-form/src/components/case/ContactDetailsAdapter.js:
```diff
@@ -0,0 +1,63 @@
+/**
+ * TODO(murilo): This file replicates some code from the hrm repo. We should implement
+ * a better solution later on.
+ */
```
Agreed!

GPaoloni commented on plugin-hrm-form/src/components/case/ViewContact.tsx:
```diff
@@ -0,0 +1,85 @@
+  const form = state[namespace][contactFormsBase].tasks[ownProps.task.taskSid];
+  const counselorsHash = state[namespace][configurationBase].counselors.hash;
+  const tempInfo = state[namespace][connectedCaseBase].tasks[ownProps.task.taskSid]
+    .temporaryCaseInfo as ViewContactType;
```
Maybe change this to type-guards inside the component (return null if `isViewContactType(temporaryCaseInfo)` is false), as there shouldn't be any case where this is called with for example a string value as `temporaryCaseInfo`, but if that ever happens this will error on mount.

murilovmachado commented:
> @murilovmachado actually my PR (#170) is relevant here because I've changed the titles on the tabbed forms in `flexUI.json`. Looks like `SearchDetails` is using a hardcoded version of those from [`SearchContact.js`](https://github.com/tech-matters/flex-plugins/blob/6aa65c9b49556779a466b33abd8f6a0f69cf4a1a/plugin-hrm-form/src/states/SearchContact.js#L142). Those need to be changed, ideally to use the same strings from `flexUI.json` if that's possible.

@nickhurlburt I've updated the PR to use the updated and localized labels.

murilovmachado commented:
Deployed to staging.

nickhurlburt commented:
Hooray!
